src/plugins/vis_types/pie/public/pie_component.tsx

```diff
@@ -186,10 +187,8 @@ const PieComponent = (props: PieComponentProps) => {
   const { visData, visParams, services, syncColors } = props;
 
   function getSliceValue(d: Datum, metricColumn: DatatableColumn) {
-    if (typeof d[metricColumn.id] === 'number' && d[metricColumn.id] !== 0) {
-      return d[metricColumn.id];
-    }
-    return Number.EPSILON;
+    const value = d[metricColumn.id];
+    return Number.isFinite(value) && value >= 0 ? value : 0;
```

Comment:
Can it be `Infinity/-Infinity`?
```suggestion
    return Math.max(0, value);
```

Reply:
@monfera as you asked about this change wdyt?

Reply:
With the formula usage, it can be I think, right? `count()/0`

Reply:
I think in formula that is handled by the expression system. Need to check that.